expyfun/stimuli/tests/test_tracker.py

```diff
@@ -72,11 +73,13 @@ def test_tracker_ud():
     tr = TrackerUD(None, 3, 1, [1, 0.5], [1, 0.5], 10, np.inf, 1,
                    change_indices=[2])
     tr.respond(True)
-    tr = TrackerUD(None, 3, 1, [1, 0.5], [1, 0.5], np.inf, 10, 1,
-                   change_indices=[2], change_rule='trials')
-    for t in np.arange(2):  # run long enough to encounter change_indices
-        tr.respond(True)
-        tr.respond(False)
+    tr = TrackerUD(None, 1, 1, [3, 1.5], [1, 0.5], np.inf, 5, 1,
+                   change_indices=[2], change_rule='trials', x_min=0, x_max=3)
+    responses = [True, False, True, True, False, True]
+    for r in responses:  # run long enough to encounter change_indices
+        tr.respond(r)
```

Comment:
We don't want un-caught warnings in tests, can you use `with warnings.catch_warnings(record=True) as w:` context manager and ensure e.g. `assert_equal(len(w), 1)` and `assert_true('invalid' in str(w[0].message))`?